tests/lib/SimpleSAML/ConfigurationTest.php

```diff
@@ -31,6 +31,13 @@ public function testLoadDefaultInstance()
      */
     public function testCriticalConfigurationError()
     {
+        try {
+            SimpleSAML_Configuration::getInstance();
+            $this->fail('Exception expected');
+        } catch (\SimpleSAML\Error\CriticalConfigurationError $var) {
+            // This exception is expected.
+        }
+        //TODO: not sure this is correct. First call to getInstance() throw exception and the 2nd works?...
```

Comment:
This is _collateral damage_. When a `CriticalConfigurationError` occurs, you are supposed to display that to the user and be done. That's the entire purpose of that exception. However, some classes will still be invoked that need a configuration object (i.e. `SimpleSAML\Logger`). Since there is no valid configuration (that's why we threw the exception in the first place), the only option is to load an emergency configuration that can be used by any class that needs it.

Here you are the victim of such behaviour, as you are capturing the exception and continuing execution after an event that's supposed to end it.